spring-pulsar/src/main/java/org/springframework/pulsar/core/DefaultPulsarProducerFactory.java

```diff
@@ -75,6 +84,9 @@ protected Producer<T> doCreateProducer(String topic, Schema<T> schema, MessageRo
 		if (messageRouter != null) {
 			producerBuilder.messageRouter(messageRouter);
 		}
+		if (producerInterceptors != null && !producerInterceptors.isEmpty()) {
```

Comment:
If you have not seen it yet, the `org.springframework.util.CollectionUtils` provides a convenience `isEmpty` that does both null/empty for us.